core/src/main/scala/filodb.core/memstore/IndexBootstrapper.scala

```diff
@@ -57,17 +280,19 @@ class IndexBootstrapper(colStore: ColumnStore) {
   def bootstrapIndexDownsample(index: PartKeyLuceneIndex,
                                shardNum: Int,
                                ref: DatasetRef,
-                               ttlMs: Long): Task[Long] = {
-
+                               ttlMs: Long,
+                               parallelism: Int = Runtime.getRuntime.availableProcessors()): Task[Long] = {
     val startCheckpoint = System.currentTimeMillis()
     val recoverIndexLatency = Kamon.gauge("shard-recover-index-latency", MeasurementUnit.time.milliseconds)
       .withTag("dataset", ref.dataset)
       .withTag("shard", shardNum)
     val start = System.currentTimeMillis() - ttlMs
     colStore.scanPartKeys(ref, shardNum)
       .filter(_.endTime > start)
-      .mapParallelUnordered(Runtime.getRuntime.availableProcessors()) { pk =>
+      .mapParallelUnordered(parallelism) { pk =>
         Task.evalAsync {
+          val dsSchema = schemas(schemaID(pk.partKey, UnsafeUtils.arayOffset)).downsample.get
+          updateDataShapeStatsIfEnabled(pk, dsSchema, shardNum)
```

Comment:
Assuming the index is already built, would this change need index to be rebuilt?